ambari-server/src/main/java/org/apache/ambari/server/upgrade/UpgradeCatalog300.java

```diff
@@ -542,6 +580,65 @@ protected void addAmbariConfigurationTable() throws SQLException {
     dbAccessor.addPKConstraint(AMBARI_CONFIGURATION_TABLE, "PK_ambari_configuration", AMBARI_CONFIGURATION_CATEGORY_NAME_COLUMN, AMBARI_CONFIGURATION_PROPERTY_NAME_COLUMN);
   }
 
+  /**
+   * Creates new tables for changed kerberos data.
+   *
+   * @throws SQLException
+   */
+  protected void upgradeKerberosTables() throws SQLException {
+    boolean keytabPrincipalTableCreated = false;
+    boolean mappingTableCreated = false;
+
+    if (!dbAccessor.tableExists(KERBEROS_KEYTAB_TABLE)) {
+      List<DBAccessor.DBColumnInfo> kerberosKeytabColumns = new ArrayList<>();
+      kerberosKeytabColumns.add(new DBAccessor.DBColumnInfo(KEYTAB_PATH_FIELD, String.class, 255, null, false));
+      kerberosKeytabColumns.add(new DBAccessor.DBColumnInfo(OWNER_NAME_FIELD, String.class, 255, null, true));
+      kerberosKeytabColumns.add(new DBAccessor.DBColumnInfo(OWNER_ACCESS_FIELD, String.class, 255, null, true));
+      kerberosKeytabColumns.add(new DBAccessor.DBColumnInfo(GROUP_NAME_FIELD, String.class, 255, null, true));
+      kerberosKeytabColumns.add(new DBAccessor.DBColumnInfo(GROUP_ACCESS_FIELD, String.class, 255, null, true));
+      kerberosKeytabColumns.add(new DBAccessor.DBColumnInfo(IS_AMBARI_KEYTAB_FIELD, Integer.class, null, 0, false));
+      kerberosKeytabColumns.add(new DBAccessor.DBColumnInfo(WRITE_AMBARI_JAAS_FIELD, Integer.class, null, 0, false));
+      dbAccessor.createTable(KERBEROS_KEYTAB_TABLE, kerberosKeytabColumns);
+      dbAccessor.addPKConstraint(KERBEROS_KEYTAB_TABLE, PK_KERBEROS_KEYTAB, KEYTAB_PATH_FIELD);
+    }
+
+    if (!dbAccessor.tableExists(KERBEROS_KEYTAB_PRINCIPAL_TABLE)) {
```

Comment:
I believe that `dbAccessor.createTable`, `dbAccessor.addPKContraint`, and  `dbAccessor.addUniqueConstraint` are safe to call without checking or this. Each ensure they are no re-adding the requested construct.   I don't think it hurts to do this though.

Reply:
Yes - they should all be safe to call without the check ahead of time.

Reply:
@rlevas @jonathan-hurley thanks for pointing to this, now I see that everything already have a check for existence. Fixed.